### PR TITLE
refactor: factor out CDP message types

### PR DESCRIPTION
**Pull request by crowlKats**
With fixes this time

### CODE REVIEW

File: cli/tools/repl/session.rs [hunk 1]
```diff
@@ -344,16 +371,29 @@ impl ReplSession {
...
+          generate_preview: None,
+          user_gesture: None,
+          await_promise: None,
+          throw_on_side_effect: None,
+          timeout: None,
+          disable_breaks: None,
+          repl_mode: Some(true),
+          allow_unsafe_eval_blocked_by_csp: None,
+          unique_context_id: None,
+        }),
```
Comment (ry) [inline cli/tools/repl/session.rs:394]: use `.. Default::default()` ?

**Comment by ry** [comment]
What is CDP messages? Why is this important?

**Comment by crowlKats** [comment]
@ry this is the exact same PR as #13501, but there were some issues, so it had to be reverted

**Review by dsherret — APPROVED** [review]
LGTM

File: cli/tools/repl/channel.rs [hunk 2]
```diff
@@ -55,17 +56,19 @@ pub struct RustylineSyncMessageSender {
 }
 
 impl RustylineSyncMessageSender {
-  pub fn post_message(
+  pub fn post_message<T: serde::Serialize>(
     &self,
     method: &str,
-    params: Option<Value>,
+    params: Option<T>,
   ) -> Result<Value, AnyError> {
```
Comment (dsherret) [inline cli/tools/repl/channel.rs:63]: I wonder if we could make the response a type parameter too? (`serde::Deserialize`)

File: cli/tests/integration/repl_tests.rs [hunk 3]
```diff
@@ -122,6 +122,17 @@ fn pty_complete_primitives() {
...
 }
 
+#[test]
+fn pty_complete_expression() {
+  util::with_pty(&["repl"], |mut console| {
+    console.write_text("Deno.\t\t");
+    console.write_text("y");
+    console.write_line("");
+    console.write_line("close();");
+    assert!(console.read_all_output().contains("Display all"));
```
Comment (bartlomieju) [inline cli/tests/integration/repl_tests.rs:132]: Could you actually assert that some of the expected fields in `Deno` namespace are in the output?

**Review by bartlomieju — APPROVED** [review]
LGTM too

**Pull request merged**